core/state/journal.go

```diff
@@ -213,6 +218,14 @@ func (ch storageChange) dirtied() *common.Address {
 	return ch.account
 }
 
+func (ch transientStorageChange) revert(s *StateDB) {
+	s.getStateObject(*ch.account).setTransientState(ch.key, ch.prevalue)
+}
+
+func (ch transientStorageChange) dirtied() *common.Address {
+	return ch.account
+}
```

Comment:
We need to look into this: I think it is incorrect to report the account as dirtied by a transient op. 

The `dirtied` address is added to the journal `dirties`, which is later used to figure out which accounts have been modified -- in the sense that the trie representation of the account is changed. But a transient op does not modify the trie representation. 

Putting this note here as a todo to investigate a bit further.

Reply:
Ah, we do need to put it in `journal.dirties` after all, because otherwise we won't call `finalise` at the end of the tx, and if we don't do that, the transient storage doesn't get reset between transactions.

Reply:
I'm not so sure that a `Storage` on the `stateObject` is the best representation of a transient storage. The state-objects live longer than a single transaction. Having the transient on the objects themselves just forces us to iterate through them at the end of each tx, just to clean up. 

The transient storage(s) are essentially a transaction-global map of `address` --> `Storage`. After each transaction, we could just nuke it out and prepare a fresh one. 

This is how we did with the accesslist. In `statedb.go`:
```
	// Per-transaction access list
	accessList *accessList
```
It uses the 'normal' journal, but it is otherwise it's own separate entity, not tied to stateobjects.

Reply:
This makes sense, it feels like it sits at a similar level of abstraction as the access list since it is applied on a per transaction basis. Thank you for the feedback, will update the design to follow this approach

Reply:
I am of the opinion that `dirtied()` should return `nil`. Setting a transient storage value does not make an account `dirty`, in the trie-sense. 
I'm open to being proven wrong

Reply:
I agree with this, a dirty account implies that it was mutated in some way